robot_skills/src/robot_skills/get_robot.py

```diff
@@ -1,32 +1,54 @@
 # System
+import importlib
+import os.path
 import sys
 
 # ROS
 import rospy
+import rospkg
 
 # Robot skills
 from .robot import Robot
 
-# ToDo: create a decent registration here
+
 ROBOTS = {}
-try:
-    from amigo_skills import Amigo
-    ROBOTS["amigo"] = Amigo
-except ImportError:
-    pass
-try:
-    from hero_skills import Hero
-    ROBOTS["hero"] = Hero
-except ImportError:
-    pass
-try:
-    from sergio_skills import Sergio
-    ROBOTS["sergio"] = Sergio
-except ImportError:
-    pass
-
-from .mockbot import Mockbot
-ROBOTS["mockbot"] = Mockbot
+
+
+def load_robots():
+    """
+    Register the robots, which are export via package.xml 'export' tag
+    """
+    rospack = rospkg.RosPack()
+    to_check = rospack.get_depends_on('robot_skills', implicit=False)
+    to_check.append("robot_skills")  # also check for robots in robot_skills
+
+    for pkg in to_check:
```

Comment:
Not much to add here except some minor issues:
* you could slightly improve readability by putting part(s) of `load_robots` in different functions, e.g., `register_robot`
* thinking about that: what happens if you register two robots with the same name? You won't notice that the behavior is not what you expected (if you are the first user unaware of the second)

Styling
```python
rospy.logerr(
    "Unable to load robot '{}' from package '{}'. Exception thrown: '{}'".format(robot, pkg, e)
)
``` 
or
```python
rospy.logerr(
    "Unable to load robot '{}' from package '{}'. Exception thrown: '{}'".format(
        robot, pkg, e
    )
)
```